pravega-sensor-collector/src/main/java/io/pravega/sensor/collector/file/FileProcessor.java

```diff
@@ -225,18 +237,36 @@ void processFile(FileNameWithOffset fileNameWithBeginOffset, long firstSequenceN
         }
     }
 
+    void moveCompletedFile(FileNameWithOffset fileNameWithBeginOffset, long endOffset, long nextSequenceNumber, Optional<UUID> txnId) throws SQLException, IOException {
+        state.addCompletedFileRecord(fileNameWithBeginOffset.fileName, fileNameWithBeginOffset.offset, endOffset, nextSequenceNumber, txnId);
+        FileUtils.moveCompletedFile(fileNameWithBeginOffset, movedFilesDirectory);
+    }
+
     void deleteCompletedFiles() throws Exception {
         final List<FileNameWithOffset> completedFiles = state.getCompletedFileRecords();
         completedFiles.forEach(file -> {
             //Obtain a lock on file
-            try(FileChannel channel = FileChannel.open(Paths.get(file.fileName), StandardOpenOption.WRITE)){
+            Path completedFilesPath = movedFilesDirectory.resolve(FileUtils.COMPLETED_FILES);
+            String completedFileName = FileUtils.createCompletedFileName(completedFilesPath, file.fileName);
+            Path filePath = completedFilesPath.resolve(completedFileName);
+            log.debug("deleteCompletedFiles: Deleting File default name:{}, and it's completed file name:{}.", file.fileName, filePath);
+            if(Files.notExists(filePath)) {
+                try {
+                    state.deleteCompletedFileRecord(file.fileName);
+                } catch (SQLException e) {
+                    throw new RuntimeException(e);
+                }
+                log.warn("deleteCompletedFiles: File {} does not exist in completed files directory.", filePath);
+                return;
+            }
+            try(FileChannel channel = FileChannel.open(filePath, StandardOpenOption.WRITE)){
                 try(FileLock lock = channel.tryLock()) {
                     if(lock!=null){
-                        Files.deleteIfExists(Paths.get(file.fileName));
-                        log.info("deleteCompletedFiles: Deleted file {}", file.fileName);
+                        Files.deleteIfExists(filePath);
```

Comment:
Can this whole block be simplified as below?
That way we don't need line 253 to 261
```
...
if(Files.notExists(filePath)) {
     log.warn("deleteCompletedFiles: File {} does not exist in completed files directory.", filePath);
} else {
   Files.deleteIfExists(filePath);
}
...

```